src/main/java/com/parsing/parsers/pdf/parsing/RotationImage.java

```diff
@@ -14,14 +15,18 @@
 @Slf4j
 public class RotationImage {
 
-    public String rotate(String filename) throws IOException {
+    public String rotate(String filename) {
         log.debug("Class RotationImage started.");
-        String result = "rotatedImage.jpg";
-        BufferedImage image = ImageIO.read(new File(filename));
-        ImageDeskew id = new ImageDeskew(image);
-        image = ImageHelper.rotateImage(image, -id.getSkewAngle());
-        ImageIO.write(image, "jpg", new File(result));
-        log.debug("Class RotationImage finished.");
-        return result;
+        try {
+            String result = "rotatedImage.jpg";
+            BufferedImage image = ImageIO.read(new File(filename));
+            ImageDeskew id = new ImageDeskew(image);
+            image = ImageHelper.rotateImage(image, -id.getSkewAngle());
+            ImageIO.write(image, "jpg", new File(result));
+            log.debug("Class RotationImage finished.");
+            return result;
+        } catch (IOException e) {
+            throw new RotationImageException("Rotation of image failed", e);
+        }
```

Comment:
close

Reply:
In the case of ImageIO, closing resources is not required because ImageIO.read and ImageIO.write do not explicitly open streams that need to be closed. They work with input and output streams that you provide and, therefore, do not require closing.

So, there is no need to add a finally block to close ImageIO. The code you provided already correctly handles exceptions and releases resources when necessary.